src/iso15118/io/socket_helper.cpp

```diff
@@ -31,10 +33,18 @@ bool get_first_sockaddr_in6_for_interface(const std::string& interface_name, soc
             continue;
         }
 
-        if (interface_name.compare(current_if->ifa_name) != 0) {
+        if (interface_name != "auto" && interface_name.compare(current_if->ifa_name) != 0) {
```

Comment:
Insted of `!= "auto"` I could also use `compare("auto") != 0`.

Reply:
I'm not sure which version would be better but I think we should use the same function to compare the strings. Otherwise it might be a bit confusing

Reply:
Done